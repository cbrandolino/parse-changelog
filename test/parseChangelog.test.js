const { readFileSync } = require('fs');
const path = require('path');
const { parseChangelogItem, parseChangelog } = require('../index.js');

describe('parseChangelog', () => {
  it('correctly parses simple semver file', () => {
    const content = readFileSync(path.join(__dirname, 'changelogExamples', 'simpleSemver.md'));

    expect(parseChangelog(content.toString())).toMatchSnapshot();
  });
});