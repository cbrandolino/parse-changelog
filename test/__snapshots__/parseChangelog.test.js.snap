// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseChangelog correctly parses simple semver file 1`] = `
Array [
  Object {
    "content": "
- Removed feature
- Added bug

",
    "date": "2009-10-10",
    "version": "v1.0",
  },
  Object {
    "content": "
- Added feature",
    "date": "2009-10-05",
    "version": "v0.2",
  },
]
`;
